src/api/schema/components/source.rs

```diff
@@ -15,17 +15,19 @@ use crate::{
 
 #[derive(Debug, Enum, Eq, PartialEq, Copy, Clone, Ord, PartialOrd)]
 pub enum SourceOutputType {
-    Any,
+    Multiple,
```

Comment:
Did you consider updating this one to also be a bitmask? Corollary: do we need two separate types to model source outputs / component inputs? Could it just be one type like `EventType`?

Reply:
This enum seems to be used only for the graphql API (I may lack knowledge on this piece) as a [filtering option](https://github.com/vectordotdev/vector/blob/master/src/api/schema/filter.rs#L50) (and filtering by output type is not implemented for anything else than source if I'm correct). 
That being said I agree this would make much more sens to rework the filter and support boolean combinator on this filter instead of having this new `Multiple` option. I'll look at that option.

Reply:
Ah, I see. Yeah, I think we could update the API to allow, rather than a single type, a list of types as input. Something like:

```
pub struct EqualityFilter<T: InputType + PartialEq + Eq> {
    pub equals: Option<T>,
    pub not_equals: Option<T>,
    pub in: Vec<T>,
    pub not_in: Vec<T>
}
```

(keeping `equals` and `not_equals` for compatibility)

I don't think GraphQL supports boolean combinators.

Reply:
I like the `in` / `not_in` semantic, I will give it a try. Anyway using a bitmask as the underlying structure for the GraphQL query param works. I also noticed we do have the `or` combinator (but not `and` at the moment, it should not by a problem to add it). With the current iteration of the PR, a working GraphQL could look like:

```graphql
query {
  sources(filter: {
    outputType:{supports:TRACE}
    or: {outputType:{supports:LOG}}
  })
  {
    edges {
      node {
        componentId
      }
    }
  }
}
```

Reply:
>  I also noticed we do have the or combinator (but not `and` at the moment, it should not by a problem to add it)

To clarify - `and` is the behaviour when multiple filter conditions are included, so there's not an explicit operator for that reason.